### ISSUE TITLE
Display ADS-B locations in forthcoming mavlink message

### ISSUE THREAD

**Issue by pgiacalo** [wiseman/mavelous#36]
Hi guys,

Happy new year and thanks again for creating mavelous.

I have a feature request...

How difficult would it be to add a feature to allow nearby 'targe't aircraft position and heading data to be input to mavelous? It would be great to have the ability to display target aircraft on the aerial map along with the flight vehicle. I'm not sure if the mavlink protocol has an api call to send this data?? 

Also, it would be useful to have vehicle altitudes displayed on the map next to each vehicle icon. And perhaps color icons could be used to indicate those aircraft above (blue), below (green), or near (red) the altitude of the flight vehicle. 

BTW, these features are being requested because they would nicely fill a GCS need for an upcoming nasa uav challenge that includes collision avoidance. 

Thanks and best regards,
Phil

**Comment by wiseman** [comment]
I don't know much about mavlink's support for multiple vehicles, but I'll
do some research.  It should be quite easy on the client side, at least.

If you know anything more about how you'll be communicating with multiple
vehicles, let me know--do you need multiple radios?

**Comment by pgiacalo** [comment]
The ads-b communication is one-way. Our vehicle will have a single receiver tuned to the ads-b channel. It will receive the ads-b signals from all nearby targets. An onboard computer will convert the signals to position and heading information for each target. This data would be then used for collision detection/avoidance calculations, and would also be sent to mavelous for display. 

The team I'm working with includes Tridge, so there's the possibility of asking him about adding future support to mavlink. Nice to hear the client side would be easy! Seems there's a possible path ahead to make this happen.

**Comment by pchickey** [comment]
On Sun, Dec 30, 2012 at 4:57 PM, Philip Giacalone
notifications@github.comwrote:

> The ads-b communication is one-way. Our vehicle will have a single
> receiver tuned to the ads-b channel. It will receive the ads-b signals from
> all nearby targets. An onboard computer will convert the signals to
> position and heading information for each target. This data would be then
> used for collision detection/avoidance calculations, and would also be sent
> to mavelous for display.
> 
> So it sounds like all of the information processing is happening on
> vehicle, and you're just depending on mavelous to show the results. I
> assume you'll be using the mavlink channel as the link to ground. So,
> you'll need to define the messages that describe the state of this target
> information, and get them added to mavlink on the ArduPlane and inside
> mavproxy via a pymavlink definition.

Once the message makes it to the ground station (mavproxy), fetching the
packet by name and displaying it is not too hard.

> The team I'm working with includes Tridge, so there's the possibility of
> asking him about adding future support to mavlink. Nice to hear the client
> side would be easy! Seems there's a possible path ahead to make this
> happen.

**Comment by pgiacalo** [comment]
Yes, processing will be handled by a flight management computer on the vehicle with a mavlink downlink. 

If possible, I'd like to get this capability in the standard mavlink api, rather than adding it for a particular implementation. Not my call but I plan to try.

**Comment by wiseman** [comment]
This does potentially relate to another idea I've been thinking about: A central site that you can send telemetry to, and request telemetry from all the other vehicles that are reporting in.  The inspiration is actually the people who [share data](http://planefinder.net/about/coverage/) from their ADS-B radios to create sites like [planefinder.net](http://planefinder.net).  [DroneOS](http://droneos.com) is a similar idea, but its development seems to have stalled.

I was thinking that mavelous could be a front end for the display of data from multiple vehicles using data from dronefinder.com or droneatlas.com or whatever it's called.

This is not an exact fit to your scenario, but it should at least guide us in designing a feature that would be useful both for you and for dronefinder.

**Comment by pgiacalo** [comment]
Yes, exactly. I completely agree. There is a generic and growing need for this ADS-B feature, so it seems like an good fit as part of the standard mavlink api. I suspect Tridge may agree, too.

**Comment by wiseman** [comment]
I forgot another example of a similar site: http://ardrone2.parrot.com/ar-drone-academy/
